sktime/utils/data_container.py

```diff
@@ -286,6 +286,24 @@ def nested_to_3d_numpy(X, a=None, b=None):
     return np.stack(
         X.iloc[a:b].applymap(lambda cell: cell.to_numpy()).apply(
             lambda row: np.stack(row), axis=1).to_numpy())
+            
+# Helper function to convert univariate data into a 2D numpy array
+# or multivariate data into a 3D numpy array
+def convert_df_into_numpy(X):
```

Comment:
we're trying to consolidate our conversion functions, could you not always use a 3d numpy array and use the above function `from_nested_to_3d_numpy`? I've not tried it but I guess for univariate data you would get `(n_instances, 1, n_timepoints)`, you can also use `np.squeeze` if you want to "squeeze" out the 2nd dimension.

Reply:
Yeah i've used from_nested_to_3d_numpy instead and that works fine, tried using np.squeeze but that caused further issues in kNN but without np.squeeze the classifier worked fine (the only thing that I was having problems with was setting the algorithm attribute in kNN to either 'ball_tree' or 'kd_tree' but 'brute' works).

Reply:
I was having issues with those algorithms anyway so I don't think that's anything I've changed

Reply:
@Multivin12 could you please raise issues about the problems you're describing re the "kd_tree" and "ball_tree"

Reply:
Yeah I'll raise an issue about this.

Reply:
Just raised the issue #328